R/plumber.R

```diff
@@ -271,16 +267,26 @@ plumber <- R6Class(
           )
         }
         self$mount("/__swagger__", PlumberStatic$new(swagger::swagger_path()))
-        swaggerUrl = paste0(host, ":", port, "/__swagger__/")
-        if (!grepl("^http://", swaggerUrl)) {
-          # must have http protocol for use within RStudio
-          # does not work if supplying "127.0.0.1:1234/route"
-          swaggerUrl <- paste0("http://", swaggerUrl)
-        }
-        message("Running the swagger UI at ", swaggerUrl, sep = "")
+
+        swaggerHost <- getOption("plumber.apiHost", host)
+        swaggerUrl = paste0(swaggerHost, ":", port, "/__swagger__/")
+        message("Running Swagger UI at  ", swaggerUrl, sep = "")
 
         # notify swaggerCallback of plumber swagger location
         if (!is.null(swaggerCallback) && is.function(swaggerCallback)) {
+          # if within the RStudio IDE...
+          if (rstudioapi::isAvailable()) {
```

Comment:
I believe we can get rid of the dependence on rstudioapi by changing to

`if (requireNamespace('rstudioapi', quietly = TRUE) && rstudioapi::isAvailable()) {`

Reply:
Asking `rstudioapi` maintainer about `rstudioapi::isAvailable` internals to avoid a dependency for a single function